### PR TITLE
Metadata header updates - INT v2.0

### LINKED ISSUE #94: Change the INT metadata header for source only information.
Due to changes to the INT 2.0 specification, we will be simplifying by removing the source MAC information from transit INT hops.

### PR DESCRIPTION
**Pull request by ghost**
#### What does this PR do?
Fixes #94.
Adds the metadata header updates for INT v2.0 with domain-specific metadata (Switch ID + Originating MAC) added at the gateway and per-hop metadata (Switch ID) added at the aggregate and core.
#### Do you have any concerns with this PR?
No
#### How can the reviewer verify this PR?
Run Mininet CI for full/gateway/aggregate test scenarios.
#### Questions:
- Have you connected this PR to the issue it resolves? Yes
- Does the documentation need an update? Yes, PR #100.
- Does this add new dependencies? No
- Have you added unit or functional tests for this PR?

### CODE REVIEW

File: trans_sec/packet/inspect_layer.py [hunk 1]
```diff
@@ -71,7 +82,7 @@ class IntMeta2(IntMeta):
     name = "INT_META_2"
 
 
-class IntMeta3(IntMeta):
+class IntMeta3(SourceMeta):
```
Comment (spisarski) [inline trans_sec/packet/inspect_layer.py]: Remove this class and just use SourceMeta in its place

File: trans_sec/device_software/receive_packets.py [hunk 2]
```diff
@@ -61,31 +61,25 @@ def __log_packet(packet, int_hops):
...
-        mac3 = None
         switch_id_3 = None
         if int_hops == 1:
-            mac1 = packet[IntMeta1].orig_mac
-            switch_id_1 = packet[IntMeta1].switch_id
+            mac1 = packet[IntMeta3].orig_mac
+            switch_id_1 = packet[IntMeta3].switch_id
         if int_hops == 2:
-            mac2 = packet[IntMeta2].orig_mac
             switch_id_2 = packet[IntMeta2].switch_id
```
Comment (spisarski) [inline trans_sec/device_software/receive_packets.py]: make sure mac1 & switch_id_1 are getting set in this block

File: trans_sec/device_software/receive_packets.py [hunk 3]
```diff
@@ -61,31 +61,25 @@ def __log_packet(packet, int_hops):
...
         if int_hops == 3:
-            mac3 = packet[IntMeta3].orig_mac
-            switch_id_3 = packet[IntMeta3].switch_id
+            switch_id_3 = packet[IntMeta1].switch_id
```
Comment (spisarski) [inline trans_sec/device_software/receive_packets.py]: make sure mac1, switch_id_1 & switch_id_2 are getting set in this block

File: trans_sec/analytics/oinc.py [hunk 4]
```diff
@@ -355,7 +355,7 @@ def __process(self, int_data):
                     'Not calling SDN as last attack notification for %s'
                     ' was only %s seconds ago',
                     attack_dict, time.time() - last_attack)
-                return False
+                return True
```
Comment (spisarski) [inline trans_sec/analytics/oinc.py]: Why was this changed to True now? Doesn't that break stuff upstream?

File: tests/trans_sec/analytics/oinc_tests.py [hunk 5]
```diff
@@ -110,7 +111,7 @@ def test_start_two_attacks(self):
             ret_val2 = self.ae.process_packet(pkt2, 0xfd)
             logger.info('Checking index - [%s] - count - [%s]',
                         index, self.ae.packet_count)
-            if index < self.ae.packet_count:
+            if index*2 < self.ae.packet_count:
```
Comment (spisarski) [inline tests/trans_sec/analytics/oinc_tests.py]: pep8 - spaces between variable, operator, and literal

**Review by spisarski — CHANGES_REQUESTED** [review]
Changes appear solid with a couple exceptions. See comments.

File: trans_sec/analytics/oinc.py (hunk 4 above)
Comment (Sanjhana-Jayagopal) [reply trans_sec/analytics/oinc.py]: @spisarski  With the _test_start_two_attacks_ we send two packets - the first packet detects an attack and ret_val1 is set to True and the second packet also detects an attack but just doesn't send the SDN controller the notification (because the notification was sent by packet 1) and so ret_val2 was being set to False. The test case _test_start_two_attacks_ was failing because ret_val1=True and ret_val2=False did not apply to any of the 'Expecting '  conditions. It seemed like a bug so changed it to True if an attack is detected irrespective of if the notification is being sent or not.

File: trans_sec/packet/inspect_layer.py [hunk 6]
```diff
@@ -71,7 +82,7 @@ class IntMeta2(IntMeta):
     name = "INT_META_2"
 
 
-class IntMeta3(IntMeta):
+class SourceIntMeta(SourceMeta):
     """
     This class represents the third INT metadata being placed onto the packets
     """
```
Comment (spisarski) [inline trans_sec/packet/inspect_layer.py]: the only other think I can think to do to improve would be to change the name of the 'new' class SourceIntMeta to something other than "INT_META_3"

**Review by spisarski — CHANGES_REQUESTED** [review]
sorry, just one minor request, plus the change will cause CI to startup again and we can see what the heck is going on with aggregate

**Review by spisarski — APPROVED** [review]
Lets do this
